core/domain/exp_services_test.py

```diff
@@ -1225,48 +1225,60 @@ def test_save_original_and_compressed_versions_of_image(self):
 
     def test_save_original_and_compressed_versions_of_image_with_dev_mode_off(
             self):
-        dev_mode_swap = self.swap(constants, 'DEV_MODE', False)
+        prod_mode_swap = self.swap(constants, 'DEV_MODE', False)
         max_resize_dimension_px_swap = self.swap(
             gae_image_services, 'MAX_RESIZE_DIMENSION_PX', 20)
         with open(os.path.join(feconf.TESTS_DATA_DIR, 'img.png')) as f:
             original_image_content = f.read()
 
-        with dev_mode_swap, max_resize_dimension_px_swap:
+        with prod_mode_swap, max_resize_dimension_px_swap:
             fs = fs_domain.AbstractFileSystem(
                 fs_domain.GcsFileSystem(
                     'exploration/%s' % self.EXPLORATION_ID))
 
-            self.assertEqual(fs.isfile('image/%s' % self.FILENAME), False)
-            self.assertEqual(
-                fs.isfile('image/%s' % self.COMPRESSED_IMAGE_FILENAME), False)
-            self.assertEqual(
-                fs.isfile('image/%s' % self.MICRO_IMAGE_FILENAME), False)
+            self.assertFalse(fs.isfile('image/%s' % self.FILENAME))
+            self.assertFalse(
+                fs.isfile('image/%s' % self.COMPRESSED_IMAGE_FILENAME))
+            self.assertFalse(fs.isfile('image/%s' % self.MICRO_IMAGE_FILENAME))
 
             exp_services.save_original_and_compressed_versions_of_image(
                 self.USER, self.FILENAME, self.EXPLORATION_ID,
                 original_image_content)
 
             # The scaling factor changes if the dimensions of the image is
             # greater than MAX_RESIZE_DIMENSION_PX.
-            width, height = gae_image_services.get_image_dimensions(
-                original_image_content)
-            new_scaling_factor = gae_image_services.MAX_RESIZE_DIMENSION_PX / (
+            width = 32
+            height = 32
+            new_scaling_factor = (
```

Comment:
This still counts as logic. Instead of doing these calculations here (and repeating the existing implementation), substitute the intended numerical values for width, height, new_width, new_height, etc. into lines 1273, 1277, 1281. Tests should focus on the outcome.

Also it looks like "compressed" and "micro" give the same width/height values? That was the sort of thing I was trying to catch. Try and use an image where they give different values. The numeric values for the three cases should be different and remain so in the future.

Reply:
I am testing the case where MAX_RESIZE_DIMENSION_PX is lesser than the image dimensions and so in this case the scaling factor will be same for micro and compressed, giving same results

Reply:
Then you should have separate tests for other cases too. Remember, test all possibilities, not just the minimum to get 100% coverage.

Reply:
Done